resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/submit/ClientSuite.scala

```diff
@@ -337,4 +337,32 @@ class ClientSuite extends SparkFunSuite with BeforeAndAfter {
     submissionClient.run()
     verify(loggingPodStatusWatcher).watchOrStop(kconf.namespace + ":driver")
   }
+
+  test("driver's confimap.metadata.namespace set to conf.namespace otherwise default") {
+    val submissionClient = new Client(
+      kconf,
+      driverBuilder,
+      kubernetesClient,
+      loggingPodStatusWatcher)
+    assert(submissionClient.createConfigs()._3.getMetadata.getNamespace === "default")
```

Comment:
```suggestion
assert(submissionClient.createConfigMap(BUILT_KUBERNETES_SPEC)._2.getMetadata.getNamespace === "default")
```

The test will also simple